File: test/test_datafile.rb
require 'lib/datafile'
require 'test/unit'

class DataFileTest < Test::Unit::TestCase

  def test_it_loads
    d = DataFile.new("data/mushroom-train.csv")
    assert_equal d.names.first, "cap-shape"
    count = 0
    d.each do |p|
      assert ['e', 'p'].include?(p.first)
      if count == 0
        assert_equal p.last['cap-shape'], 'x'
      end
      count += 1
    end
    assert_equal count, 6124
  end

end

File: tests/test_datafile.py
from bayesbot.datafile import DataFile

import unittest

class DataFileTest(unittest.TestCase):

    def test_it_loads(s):
        d = DataFile("data/mushroom.csv")
        s.assertEqual(d.features[0], "cap-shape")
        for klass, features in d.points():
            s.assertIn(klass, ["p", "e"])
            s.assertIn(features["cap-shape"], ["f", "x", "b", "s", "k", "c"])
        s.assertEqual(d.count, 8124)
